### DISCUSSION TITLE
Next.js catch-all route returns 404s in production (@now/next)

### DISCUSSION THREAD

**Discussion by good-idea** [vercel/vercel#4764] [Help]
I'm posting a lot here today...

Another issue I'm having is with a Next.js catchall route. I have these pages defined:

```
.
├── _app.tsx
├── _document.tsx
├── _error.tsx
├── account.tsx
├── dashboard
│   ├── [...dashboardPage].tsx
│   └── index.tsx
├── index.tsx
├── login.tsx
├── register.tsx
└── templates
    ├── [templateSlug].tsx
    └── index.tsx
```

The routes I'm having trouble with are anything past `/dashboard`, i.e. `/dashboard/my-account`, etc. Trying to load them in production gives me a 404. Everything works fine when developing locally.

(This is a "client-side only" route, since the viewer will need to be logged in, etc, and it doesn't need to have SSR or anything.)

The original `[...dashboardPage].tsx` file looks like this:

```
import * as React from 'react'
import dynamic from 'next/dynamic'

const DashboardView = dynamic(
  async () => {
    const mod = await import('@myrepo/ui/src/views/DashboardView')
    return mod.DashboardView
  },
  {
    ssr: false,
  },
)

const DashboardPage = () => <DashboardView />

export default DashboardPage
```

But, even when I dumb it down to this:

```
import * as React from 'react'

const DashboardPage = () => {
  return <p>Your Dashboard</p>
}

export default DashboardPage
```

I still get the same result. `/dashboard` will load, but `/dashboard/anything-else` returns a 404.

**Comment by paulogdm** [comment]
Are you able to give us a reproducible repository?

**Comment by good-idea** [comment]
Yeah, here you go:

https://github.com/good-idea/next-catchall-test

It looks like it's actually an issue with dynamic routes, not specific to catch-all routes. I added another dynamic route, this is what my pages directory looks like now:

```
├── [param].tsx
├── index.tsx
└── path
    ├── [...catchAll].tsx
    └── index.tsx
```

Deployed:

https://next-test.good-idea.vercel.app/path (works)
https://next-test.good-idea.vercel.app/path/param (dynamic catchall route, 404)
https://next-test.good-idea.vercel.app/param (dynamic route, 404)

This works locally with `vercel dev`.
